src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseImportService.java

```diff
@@ -89,10 +94,13 @@ public ProgrammingExercise importProgrammingExerciseBasis(final ProgrammingExerc
         setupTestRepository(newExercise, projectKey);
         programmingExerciseService.initParticipations(newExercise);
 
-        // Hints and test cases
+        // Hints, test cases and static code analysis categories
         exerciseHintService.copyExerciseHints(templateExercise, newExercise);
         programmingExerciseRepository.save(newExercise);
         importTestCases(templateExercise, newExercise);
+        if (Boolean.TRUE.equals(templateExercise.isStaticCodeAnalysisEnabled())) {
+            importStaticCodeAnalysisCategories(templateExercise, newExercise);
```

Comment:
Just to make sure: The `@Transactional` annotation ensures that the imported SCA categories get saved into the DB although `programmingExerciseRepository.save(newExercise);` is only called before setting the categories, right?

Reply:
I honestly don't know why we use a Transactional here. From a DB perspective, the table storing the programming exercise doesn't have information about the categories. The reference (`exercise_id`) resides in the category table as usual in a 1:n relationship. The `importStaticCodeAnalysisCategories` uses `newExercise` which already contains the `exercise_id` after `programmingExerciseRepository.save(newExercise)` was called to create new categories with the right relation one by one.